doc/source/whatsnew/v2.1.0.rst

```diff
@@ -160,6 +161,7 @@ Other enhancements
 - :meth:`DataFrame.stack` gained the ``sort`` keyword to dictate whether the resulting :class:`MultiIndex` levels are sorted (:issue:`15105`)
 - :meth:`DataFrame.unstack` gained the ``sort`` keyword to dictate whether the resulting :class:`MultiIndex` levels are sorted (:issue:`15105`)
 - :meth:`DataFrameGroupby.agg` and :meth:`DataFrameGroupby.transform` now support grouping by multiple keys when the index is not a :class:`MultiIndex` for ``engine="numba"`` (:issue:`53486`)
+- :meth:`GroupBy.first` if series contains only NA values (which might be NaN), return the first NA value, else return np.nan (see `PR <https://github.com/pandas-dev/pandas/pull/53970>`_)
```

Comment:
Not sure what this is decribing

Reply:
This is describing a change of behavior of the ``GroupBy.first`` helper function to not assume that np.nan is a valid NA value for the EA in question.  The idea of the change (which, if valid, should also be applied to ``GroupBy.last``) is that if we cannot find any non-NA values in the array, but if there are NA values, the return value should be the first (or last) NA value in the array.  I did this because I thought that my EA could only deal with NaN values that were UFloats (having a nominal value and a std_dev value).  But as I traced through the code and observed things more closely, I see that the EA does have tolerance for np.nan as an element among UFloats, giving np.nan as the nominal value and zero as the std_dev.  So I will remove this change (by being less strict in some of my own assumptions in checking EA consistency).